services/friendbot/init_friendbot.go

```
@@ -13,24 +15,71 @@ func initFriendbot(
 	networkPassphrase string,
 	horizonURL string,
 	startingBalance string,
+	numMinions int,
 ) *internal.Bot {
-
-	if friendbotSecret == "" || networkPassphrase == "" || horizonURL == "" || startingBalance == "" {
+	if friendbotSecret == "" || networkPassphrase == "" || horizonURL == "" || startingBalance == "" || numMinions < 0 {
 		return nil
 	}
 
 	// ensure its a seed if its not blank
 	strkey.MustDecode(strkey.VersionByteSeed, friendbotSecret)
 
+	hclient := &horizon.Client{
+		URL:     horizonURL,
+		HTTP:    http.DefaultClient,
+		AppName: "friendbot",
+	}
+
+	minions := make([]internal.Minion, numMinions)
+	for i := 0; i < numMinions; i++ {
+		minionSecret, err := createNewAccount(friendbotSecret, networkPassphrase, hclient)
+		if err != nil {
+			return nil
```

this should return a wrapped error, the return type for this method should be updated to return the bot and an error.